### DISCUSSION TITLE
How to connect two different networks

### DISCUSSION THREAD

**Discussion by imrank03** [libp2p/rust-libp2p#2789] [Q&A]
Hi, I am able to connect with local address, I need to know how to connect with different network or address , for example my friend  is connected different network and he is different location and I want to connect him.

Any example available? or any suggestions

**Comment by mxinden** [comment]
You would use their public IP address. In case they don't have a public IP address you would need to look into hole punching.

https://blog.ipfs.io/2022-01-20-libp2p-hole-punching/

**Reply by imrank03** [reply]
Thanks for your response, if they have a public IP address, I need to change the local IP to public in the chat app, correct me if I'm wrong, because I am a newcomer.
